Tests/Engine/DataFeeds/Enumerators/FillForwardEnumeratorTests.cs

```diff
@@ -1238,6 +1240,297 @@ public void FillsForwardDaylightSavingTime(DateTime reference)
+                        {
+                            Time = reference,
+                            Value = 1,
+                            Period = dataResolution,
+                            Volume = 100
+                        }
+                    }),
+                new BaseDataCollection(
+                    reference.AddDays(1),
+                    reference.AddDays(1).Add(dataResolution),
+                    spy,
+                    new List<BaseData>{new TradeBar
+                        {
+                            Time = reference.AddDays(1),
+                            Value = 2,
+                            Period = dataResolution,
+                            Volume = 100
+                        }
+                    }),
+                new Delisting(spy, expiry.Date, 100, DelistingType.Warning),
+                new BaseDataCollection(
+                    reference.AddDays(12),
+                    reference.AddDays(12).Add(dataResolution),
+                    spy,
+                    new List<BaseData>
+                    {
+                        new TradeBar
+                        {
+                            Time = reference.AddDays(12),
+                            Value = 1,
+                            Period = dataResolution,
+                            Volume = 100
+                        }
+                    }),
+                new Delisting(spy, delisted, 100, DelistingType.Delisted)
+            }.ToList();
+
+            var enumerator = data.GetEnumerator();
+
+            var fillForwardEnumerator = new FillForwardEnumerator(
+                enumerator,
+                exchange,
+                Ref.Create(TimeSpan.FromDays(1)),
+                false,
+                data.Last().EndTime,
+                dataResolution,
+                dataTimeZone,
+                data.First().EndTime);
+
+            // Fast forward 2014.06.05 - 06
+            while (fillForwardEnumerator.MoveNext())
+            {
+                Assert.IsTrue(fillForwardEnumerator.MoveNext());
+                if (fillForwardEnumerator.Current.Time.Day == 7)
+                {
+                    break;
+                }
+            }
+
+            var dateSet = new HashSet<DateTime>();
+            while (fillForwardEnumerator.MoveNext())
+            {
+                Assert.NotNull(fillForwardEnumerator.Current);
+                Assert.AreEqual(
+                    fillForwardEnumerator.Current.DataType != MarketDataType.Auxiliary,
+                    fillForwardEnumerator.Current.IsFillForward || fillForwardEnumerator.Current.Time == data[3].Time);
+                if (fillForwardEnumerator.Current.IsFillForward)
+                {
+                    dateSet.Add(fillForwardEnumerator.Current.Time.Date);
+                }
+            }
+
+            Assert.AreEqual(
+                (int)(data.Last().EndTime - data[1].EndTime).TotalDays + 1,
```

Comment:
Is there a reason to get the difference in days from the second object in `data` and then add 1?

Why don't you simply get the difference using the first element (index 0)?

Reply:
oh. it's all about interpretation :D
If I put `data[0]` or `data[1].Time` you can interpret that that we have doubled data in the start, but in fact we should receive Aux and not-Aux data on last date (because of minute resolution). But you are right, in both case output is the same.